### PR TITLE
Add documentation section for 'mapi_instance_OP_failed' metrics

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Added section to the documentation explains typical use-cases these metrics should cover.

### CODE REVIEW

File: docs/dev/metrics.md [hunk 1]
```diff
@@ -88,6 +88,28 @@ In addition, Prometheus provides some default metrics about the internal state
 of the running process and the metric collection. You can find more information
 about these metric names and their labels through the following links:
 
+## Machine API error rate for provider
+
+These metrics should help you collect and show errors, returned from the cloud API.
```
Comment (JoelSpeed) [inline docs/dev/metrics.md:93]: Can be more direct here
```suggestion
These values show errors returned by cloud provider APIs.
```

File: docs/dev/metrics.md [hunk 2]
```diff
@@ -88,6 +88,28 @@ In addition, Prometheus provides some default metrics about the internal state
...
+
+Such metrics represent the cause for:
+- invalid provider configurations yet unknown to MAO static validation
+- timeouts and errors on the server side
```
Comment (JoelSpeed) [inline docs/dev/metrics.md:97]: Suggestion to try better fit the style of the rest of the document, WDYT?
```suggestion
For example:
- Invalid provider configurations not identified by Machine API static validation
- Timeouts and internal errors on the server side
```

File: docs/dev/metrics.md [hunk 3]
```diff
@@ -88,6 +88,28 @@ In addition, Prometheus provides some default metrics about the internal state
...
+
+In addition they are counting error occurence over a defined timespan.
```
Comment (JoelSpeed) [inline docs/dev/metrics.md:99]: Not sure what you're trying to say here, do you mean that these can be chopped up in prometheus to look at errors over a given period? What kind of metric is this? A counter?

**Review by JoelSpeed — COMMENTED** [review]
Thanks for adding this, added a couple of suggestions to make the new section try to feel more uniform with the rest of the doc, let me know what you think

File: docs/dev/metrics.md (hunk 3 above)
Comment (Danil-Grigorev) [reply docs/dev/metrics.md:99]: Just a note to the fact, that we initially planned those to be used for API failure rate. Yes, it is a counter.

Comment (JoelSpeed) [reply docs/dev/metrics.md:99]: Ok, I'd drop this sentence, the rate etc can all be calculated/extrapolated in prometheus from any metric really

**Review by JoelSpeed — APPROVED** [review]
/approve

**Review by alexander-demichev — APPROVED** [review]
/lgtm

**Review by michaelgugino — APPROVED** [review]
/lgtm

File: docs/dev/metrics.md [hunk 4]
```diff
@@ -88,6 +88,26 @@ In addition, Prometheus provides some default metrics about the internal state
 of the running process and the metric collection. You can find more information
 about these metric names and their labels through the following links:
 
+## Machine API error rate for provider
```
Comment (elmiko) [inline docs/dev/metrics.md:91]: the placement of this section cuts the "Metrics about the Prometheus collectors" in half. this new section should be moved to line 73 at the least.

**Review by elmiko — CHANGES_REQUESTED** [review]
this new section cuts another section in half
/hold

**Comment by Danil-Grigorev** [comment]
/hold cancel
@elmiko good catch! PTAL

**Review by elmiko — APPROVED** [review]
thanks Danil!
/lgtm

**Pull request merged**